crates/oxc_transformer/src/es2018/async_generator_functions/mod.rs

```diff
@@ -164,22 +164,14 @@ impl<'a, 'ctx> Traverse<'a> for AsyncGeneratorFunctions<'a, 'ctx> {
         }
     }
 
-    fn exit_method_definition(
-        &mut self,
-        node: &mut MethodDefinition<'a>,
-        ctx: &mut TraverseCtx<'a>,
-    ) {
-        let function = &mut node.value;
-        if function.r#async && function.generator && !function.is_typescript_syntax() {
-            self.executor.transform_function_for_method_definition(function, ctx);
-        }
-    }
-
     fn enter_function(&mut self, func: &mut Function<'a>, _ctx: &mut TraverseCtx<'a>) {
         self.stack.push(func.r#async && func.generator);
     }
 
-    fn exit_function(&mut self, _func: &mut Function<'a>, _ctx: &mut TraverseCtx<'a>) {
+    fn exit_function(&mut self, func: &mut Function<'a>, ctx: &mut TraverseCtx<'a>) {
+        if func.r#async && func.generator && !func.is_typescript_syntax() {
+            self.executor.transform_function_for_method_definition(func, ctx);
+        }
```

Comment:
Do you need to check if this function is a method?

```suggestion
        if func.r#async
            && func.generator
            && !func.is_typescript_syntax()
            && matches!(ctx.parent(), Ancestor::MethodDefinitionValue(_))
        {
            self.executor.transform_function_for_method_definition(func, ctx);
        }
```